src/lib/tls/tls13/tls_extensions_key_share.cpp

```diff
@@ -234,12 +255,47 @@ class Key_Share_ClientHello
    public:
       Key_Share_ClientHello(TLS_Data_Reader& reader, uint16_t /* extension_size */)
          {
+         // This construction is a crutch to make working with the incoming
+         // TLS_Data_Reader bearable. Currently, this reader spans the entire
+         // Client_Hello message. Hence, if offset or length fields are skewed
+         // or maliciously fabricated, it is possible to read further than the
+         // bounds of the current extension.
+         // Note that this aplies to many locations in the code base.
+         //
+         // TODO: Overhaul the TLS_Data_Reader to allow for cheap "sub-readers"
+         //       that enforce read bounds of sub-structures while parsing.
```

Comment:
Can you open an issue about this? It is definitely a weak point with regards to robustness, especially in extension parsing.

Reply:
I suppose [this](https://github.com/randombit/botan/pull/3144) obsoletes the need for an issue.

Reply:
There are some bullet points (under "Potential Future Work") for internal improvements in [the original TLS client PR](https://github.com/randombit/botan/pull/2922) which is also referenced in [the TLS 1.3 "current state" issue](https://github.com/randombit/botan/issues/2990). Admittedly, that's somewhat hidden. Though, I figured a bullet point list would be better than cluttering the Issues with tickets at this point.